packages/sdk/wallets/wallet-walletconnect/test/e2e.test.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  ensureLeading0x,
+  eqAddress,
+  privateKeyToAddress,
+  privateKeyToPublicKey,
+  trimLeading0x,
+} from '@celo/utils/lib/address'
+import { computeSharedSecret as computeECDHSecret } from '@celo/utils/lib/ecdh'
+import { ECIES } from '@celo/utils/lib/ecies'
+import { verifyEIP712TypedDataSigner, verifySignature } from '@celo/utils/src/signatureUtils'
+import { recoverTransaction } from '@celo/wallet-base'
+import Web3 from 'web3'
+import { WalletConnectWallet } from '../src'
+import { getTestWallet, testAddress, testPrivateKey } from './in-memory-wallet'
+
+const CHAIN_ID = 44378
+const TYPED_DATA = {
+  types: {
+    EIP712Domain: [
+      { name: 'name', type: 'string' },
+      { name: 'version', type: 'string' },
+      { name: 'chainId', type: 'uint256' },
+      { name: 'verifyingContract', type: 'address' },
+    ],
+    Person: [
+      { name: 'name', type: 'string' },
+      { name: 'wallet', type: 'address' },
+    ],
+    Mail: [
+      { name: 'from', type: 'Person' },
+      { name: 'to', type: 'Person' },
+      { name: 'contents', type: 'string' },
+    ],
+  },
+  primaryType: 'Mail',
+  domain: {
+    name: 'Ether Mail',
+    version: '1',
+    chainId: 1,
+    verifyingContract: '0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC',
+  },
+  message: {
+    from: {
+      name: 'Cow',
+      wallet: '0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826',
+    },
+    to: {
+      name: 'Bob',
+      wallet: '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB',
+    },
+    contents: 'Hello, Bob!',
+  },
+}
+
+describe('e2e tests', () => {
+  const wallet = new WalletConnectWallet({
+    name: 'Example Dapp',
+    description: 'Example Dapp for WalletConnect',
+    url: 'https://example.org/',
+    icons: ['https://example.org/favicon.ico'],
+  })
+  const testWallet = getTestWallet()
+
+  beforeAll(async () => {
+    wallet.getUri().then((uri) => testWallet.init(uri))
```

Comment:
Do you not like async/await :P

Reply:
See comment on Slack for my dilemma on API design @nambrot